clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/HostLevelIngestionStats.java

```diff
@@ -279,29 +279,51 @@ public HostLevelIngestionStats(
     this.totalHybridRecordsConsumedByRegionId =
         Collections.unmodifiableList(Arrays.asList(tmpTotalHybridRecordsConsumedByRegionId));
 
-    // Stats which are per-store only:
+    // Register an aggregate metric for disk_usage_in_bytes metric
+    final String DISK_USAGE_IN_BYTES_METRIC = "disk_usage_in_bytes";
+    if (isTotalStats()) {
+      registerSensor(
+          DISK_USAGE_IN_BYTES_METRIC,
+          new Gauge(
+              () -> ingestionTaskMap.values()
+                  .stream()
+                  .mapToLong(task -> task.getStorageEngine().getCachedStoreSizeInBytes())
+                  .sum()));
+    } else {
+      registerSensor(
+          DISK_USAGE_IN_BYTES_METRIC,
+          new Gauge(
+              () -> ingestionTaskMap.values()
+                  .stream()
+                  .filter(task -> task.getStoreName().equals(storeName))
+                  .mapToLong(task -> task.getStorageEngine().getStoreSizeInBytes())
+                  .sum()));
+    }
+    // Register an aggregate metric for rmd_disk_usage_in_bytes metric
+    final String RMD_DISK_USAGE_IN_BYTES_METRIC = "rmd_disk_usage_in_bytes";
+    if (isTotalStats()) {
+      registerSensor(
+          RMD_DISK_USAGE_IN_BYTES_METRIC,
+          new Gauge(
+              () -> ingestionTaskMap.values()
+                  .stream()
+                  .mapToLong(task -> task.getStorageEngine().getCachedRMDSizeInBytes())
+                  .sum()));
+    } else {
+      registerSensor(
+          RMD_DISK_USAGE_IN_BYTES_METRIC,
+          new Gauge(
+              () -> ingestionTaskMap.values()
+                  .stream()
+                  .filter(task -> task.getStoreName().equals(storeName))
+                  .mapToLong(task -> task.getStorageEngine().getRMDSizeInBytes())
+                  .sum()));
+    }
```

Comment:
Nitpick: I think this code could be a bit less repetitive if introducing a new function. Something like:

```
  private void registerDiskSizeMetric(
      String storeName,
      String metricName,
      Map<String, StoreIngestionTask> ingestionTaskMap,
      ToLongFunction<AbstractStorageEngine> sizeGetter) {
    Predicate<? super StoreIngestionTask> filter = isTotalStats() ? t -> true : t -> t.getStoreName().equals(storeName);
    registerSensor(
        metricName,
        new Gauge(() -> ingestionTaskMap.values()
            .stream()
            .filter(filter)
            .map(task -> task.getStorageEngine())
            .mapToLong(sizeGetter)
            .sum()));
  }
```

Then the code here in the constructor could become:

```
    registerDiskSizeMetric(
        storeName,
        "disk_usage_in_bytes",
        ingestionTaskMap,
        isTotalStats() ? AbstractStorageEngine::getCachedStoreSizeInBytes : AbstractStorageEngine::getStoreSizeInBytes);

    registerDiskSizeMetric(
        storeName,
        "rmd_disk_usage_in_bytes",
        ingestionTaskMap,
        isTotalStats() ? AbstractStorageEngine::getCachedRMDSizeInBytes : AbstractStorageEngine::getRMDSizeInBytes);
```

Reply:
I updated the code a little to remove some of duplicate code and I was trying to avoid creating another function to make it easier to read.
Let me know whether it is good enough or not.